src/utils.ts

```diff
@@ -106,3 +106,12 @@ export function useMountedState(): () => boolean {
 
   return get;
 }
+
+export function jestIsolateModules<T>(f: () => T) {
+  return new Promise((resolve, reject) => {
+    jest.isolateModules(() => {
```

Comment:
We will probably have to check if `jest is defined` and have an approach in place that could work across different testing frameworks? Perhaps others do support manual modification of `require.cache` and it wont be that complicated anyway.

Reply:
Yes, 100% agree. I'd love to keep this more or less framework agnostic.

This means providing a `isolateModules` equivalent that works with vanilla `require`. By now I found a few packages to bypass require's cache, but they don't consider nested requires:

- https://www.npmjs.com/package/requirefresh
- https://www.npmjs.com/package/import-fresh

We might search better or try our own implementation (if feasible). Still we wouldn't be able to support out of the box other test frameworks which provide their own `require` reimplementation.

In order for the strategy taken in this PR to succeed, we need to wait until `jest` publishes its next release with [fixed `isolatedModules`](https://github.com/facebook/jest/commit/e5a84d92fc906a5bb140f9753b644319cea095da).

In the meanwhile I might try to patch Jest with `patch-package`.